### DISCUSSION TITLE
Append for FieldArray ignoring defaultValue

### DISCUSSION THREAD

**Discussion by Hengry** [react-hook-form/react-hook-form#9114] [Q&A]
Is it possible to append a `FieldArray` with all fields in undefined when the `FieldArray` is reset by specific value?

[sandbox](https://codesandbox.io/s/react-hook-form-usefieldarray-forked-0d3vnv?file=/src/index.js)
It can reproduced by clicking `RESET` -> `DELETE` -> `APPEND`.

I know the document suggest append should accompany with defaultValue, and it can be achieved by `append({ firstName: '', gender: null })`.
While `null` and `undefined` has different meaning for me, so I would like to let gender set to undefined.

I've tried `append({ firstName: '', gender: undefined })` and `append({})` but both of them are failed.

Is there any suggestion? Thanks a lot.

**Comment by bluebill1049** [comment]
https://react-hook-form.com/api/usefieldarray

> When you append, prepend, insert and update the field array, the obj can't be an empty object rather need to supply all your input's defaultValues.

<img width="901" alt="Screen Shot 2022-09-30 at 8 22 12 am" src="https://user-images.githubusercontent.com/10513364/193152579-650fed97-052d-4242-989f-cab3485a9967.png">

**Reply by Hengry** [reply]
Thank you for your answer.
While as my question shows, I have saw the document before so I know my requirement does not match the suggestion.
But I still need to append with undefined defaultValues.
Is it means my requirement is not feasible?

Since undefined, '', null have different meaning for me, I need something like append({ firstName: undefined, lastName: undefined }) to discriminate from null or ''.

Or maybe any suggestion to work around? Thank you.
